stripe/src/test/java/com/stripe/android/FileUploadRequestTest.kt

```diff
@@ -0,0 +1,94 @@
+package com.stripe.android
+
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import com.stripe.android.model.StripeFileParams
+import com.stripe.android.model.StripeFilePurpose
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class FileUploadRequestTest {
+    private val context: Context by lazy {
+        ApplicationProvider.getApplicationContext<Context>()
+    }
+    private val fileFactory: FileFactory by lazy {
+        FileFactory(context)
+    }
+
+    @Test
+    fun writeBody_shouldWriteExpectedNumberOfBytes() {
+        val request = FileUploadRequest(
+            StripeFileParams(
+                file = fileFactory.create(),
+                purpose = StripeFilePurpose.IdentityDocument
+            ),
+            OPTIONS,
+            boundary = "5955816017232305695"
+        )
+
+        FakeOutputStream().use {
```

Comment:
if all you need is the written bytes, it might be easier to just use a plain old `ByteArrayOutputStream` and call `size()` on it, no need to roll your own fake version :)